src/promptflow-evals/promptflow/evals/evaluators/_content_safety/common/content_safety_base.py

```diff
@@ -33,7 +28,7 @@ def __init__(self, metric: EvaluationMetrics, project_scope: dict, credential=No
         self._project_scope = project_scope
         self._credential = credential
 
-    def __call__(self, *, question: str, answer: str, **kwargs):
+    async def __call__(self, *, question: str, answer: str, **kwargs):
```

Comment:
Would it make more sense to add the use of `async_run_allowing_running_loop` here instead of in the child classes? It seems like the child `__call__` implementations could be removed as along as `self._async_evaluator` always exists.

Reply:
Good idea, but unfortunately, `async_run_allowing_running_loop` needs to be called inside a synchronous function. Therefore, we have to leave it in the synchronous subclasses.